document new VoidFunctionComponent type (#269)
typescript-cheatsheets/react: docs/basic/getting-started/function-components.md

}) => <div title={title}>{children}</div>;
```

<details>
<summary>

> As of [@types/react PR #46643](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46643), you can use a new `React.VoidFunctionComponent` or `React.VFC` type if you wish to declare a component that explicitly does not have children.

Reviewer: ```suggestion
As of [@types/react PR #46643](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46643), you can use a new `React.VoidFunctionComponent` or `React.VFC` type if you wish to declare the accepted `children` explicitly.
```

`VoidFunctionComponent` can act as a migration path (for a planned breaking change) where you replace all `FunctionComponent` with `VoidFunctionComponent` and see what breaks. Might add that as well and mention that it's temporary as Ferdy said.

}) => <div title={title}>{children}</div>;
```

<details>
<summary>

> As of [@types/react PR #46643](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46643), you can use a new `React.VoidFunctionComponent` or `React.VFC` type if you wish to declare the accepted `children` explicitly. This is an interim solution until the next major version of the type defs (where VoidFunctionComponent will be deprecated and FunctionComponent will by default accept no children).

Reviewer: We should replace the PR link with the types version once released. PR links aren't helpful if you don't know what you're looking for.

Also:
```suggestion
As of [@types/react PR #46643](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46643), you can use a new `React.VoidFunctionComponent` or `React.VFC` type if you wish to declare the accepted `children` explicitly. This is an interim solution until FunctionComponent will by default accept no children (planned for React 18).
```
We're most likely not doing any types breaking changes in 17. See https://github.com/DefinitelyTyped/DefinitelyTyped/issues/46691#issuecomment-672840456 for more context.
